webapp/schemas.py

```diff
@@ -382,17 +374,41 @@ class CVEAPISchema(CVESchema):
     package_statuses = List(Nested(CvePackage), data_key="packages")
     notices_ids = List(
         String(validate=Regexp(r"(USN|LSN)-\d{1,5}-\d{1,2}")),
-        data_key="notices",
     )
 
 
-class CVEsAPISchema(Schema):
+class NoticeAPIDetailedSchema(NoticeAPISchema):
     cves = List(Nested(CVEAPISchema))
+
+
+class CVEAPIDetailedSchema(CVEAPISchema):
+    notices = List(Nested(NoticeAPISchema))
```

Comment:
@albertkol am I correct in assuming that basic NoticeAPISchema instead of NoticeAPIDetailedSchema prevents recursive CVE<->USN relationships where USN includes a full CVE response which include that USN full response etc.?

Reply:
@blackboxsw exactly. This way I prevent potential long rabbit holes cycles of returning full payloads of full payloads.

For example:
- Let's say we call `/notices/USN-ABCD-1.json`
- `notice.cves` will return full payloads of `CVE-ABCD-1` and `CVE-ABCD-2`.
- `notices.cves_ids` will return `["CVE-ABCD-1", "CVE-ABCD-2"]`
- But the payload of `CVE-ABCD-1` and `CVE-ABCD-2` will not have `notices` at all. It will just have `notices_ids`.

Hope that makes sense. If there is a way we could improve it to be more helpful to the `ua fix` tool, we should discuss it.

Also we were talking a while about about `notice.related_notices`. A way to get that now would be to iterate through all the `notice.cves` -> `cve.notices_ids` and you would have that.